core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -205,6 +208,8 @@ public void onReady() {}
    * useful debug information. Both argument being null may log warnings and result in suboptimal
    * performance. Also note that the provided information will not be sent to the server.
    *
+   * <p>This method is safe to call from multiple threads without external synchronization.
```

Comment:
I'm not sure this is threadsafe.  request causes events to be fired, but doesn't change the state of the object, where this does.  The only known implementation of ClientCall is not threadsafe for cancel (i.e. `ClientCallImpl`, which has unsafe access to `cancelCalled`).   @ejona86  can comment on whether this method _should_ be thread safe.

Reply:
iiuc, cancel() is called directly when someone with a handle to a ListenableFuture (e.g. from a FutureStub) calls ListenableFuture.cancel(true). This could be on any thread. 

https://github.com/grpc/grpc-java/blob/master/stub/src/main/java/io/grpc/stub/ClientCalls.java#L483

If the current implementation's cancel() is not thread-safe, then I think it needs to be fixed.

Reply:
That sounds plausible.    We probably didn't notice due to hardly ever using the ListenableFuture stubs.   In any case, the docs for `cancel()` cannot be updated before the implementation.

Reply:
Makes sense. I will confirm this and then create a tracking issue if correct. In the meantime, removed comments about cancel().

Reply:
Created #4547 to track.